syncd/MdioIpcServer.cpp

```diff
@@ -0,0 +1,469 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <ctype.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/select.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <time.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+#include "MdioIpcServer.h"
+
+#include "meta/sai_serialize.h"
+
+#include "swss/logger.h"
+
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <vector>
+
+#define SYNCD_IPC_SOCK_SYNCD  "/var/run/sswsyncd"
+#define SYNCD_IPC_SOCK_HOST   "/var/run/docker-syncd"
+#define SYNCD_IPC_SOCK_FILE   "mdio-ipc"
+#define SYNCD_IPC_BUFF_SIZE   256   /* buffer size */
+
+#define CONN_TIMEOUT        30     /* sec, connection timeout */
+#define CONN_MAX            18     /* max. number of connections */
+
+#ifndef COUNTOF
+#define COUNTOF(x)          ((int)(sizeof((x)) / sizeof((x)[0])))
+#endif
+
+using namespace syncd;
+
+bool MdioIpcServer::m_syncdContext = true;
+
+typedef struct syncd_mdio_ipc_conn_s
+{
+    int    fd;
+    time_t timeout;
+} syncd_mdio_ipc_conn_t;
+
+MdioIpcServer::MdioIpcServer(
+        _In_ std::shared_ptr<sairedis::SaiInterface> vendorSai,
+        _In_ int globalContext):
+    m_vendorSai(vendorSai),
+    m_switchRid(SAI_NULL_OBJECT_ID),
+    m_taskThread(),
+    m_taskAlive(0)
+{
+    SWSS_LOG_ENTER();
+
+    /* globalContext == 0 for syncd, globalContext > 0 for gbsyncd */
+    MdioIpcServer::m_syncdContext = (globalContext == 0);
+}
+
+MdioIpcServer::~MdioIpcServer()
+{
+    SWSS_LOG_ENTER();
+
+    m_taskAlive = 0;
+    if(m_taskThread.joinable()) m_taskThread.join();
```

Comment:
dont put code in the same line as if() please move to separate line

Reply:
Changed according to your comment.

Reply:
Changed according to your comment.